main.go

```diff
@@ -0,0 +1,543 @@
+	}
+	return json.Unmarshal(jsonPayload, &task.Payload)
+}
+
+func (task *TaskRun) reportException() error {
+	ter := queue.TaskExceptionRequest{Reason: json.RawMessage(`"malformed-payload"`)}
+	tsr, callSummary := Queue.ReportException(task.TaskId, strconv.FormatInt(int64(task.RunId), 10), &ter)
+	if callSummary.Error != nil {
+		log.Printf("Not able to report exception for task %v:\n%v", task.TaskId, callSummary.Error)
+		return callSummary.Error
+	}
+	task.TaskClaimResponse.Status = tsr.Status
+	log.Println(task.String())
+	return nil
+}
+
+func (task *TaskRun) reportFailed() error {
+	tsr, callSummary := Queue.ReportFailed(task.TaskId, strconv.FormatInt(int64(task.RunId), 10))
+	if callSummary.Error != nil {
+		log.Printf("Not able to report failed completion for task %v:\n%v", task.TaskId, callSummary.Error)
+		return callSummary.Error
+	}
+	task.TaskClaimResponse.Status = tsr.Status
+	log.Println(task.String())
+	return nil
+}
+
+func (task *TaskRun) reportCompleted() error {
+	log.Printf("Command finished successfully!")
+	tsr, callSummary := Queue.ReportCompleted(task.TaskId, strconv.FormatInt(int64(task.RunId), 10))
+	if callSummary.Error != nil {
+		log.Printf("Not able to report successful completion for task %v:\n%v", task.TaskId, callSummary.Error)
+		return callSummary.Error
+	}
+	task.TaskClaimResponse.Status = tsr.Status
+	log.Println(task.String())
+	return nil
+}
+
+func (task *TaskRun) run() error {
+
+	fmt.Println("Running task!")
+	fmt.Println(task.String())
+	task.prepEnvVars()
+	cmd := exec.Command(task.Payload.Command[0], task.Payload.Command[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+	log.Printf("Waiting for command to finish...")
+	// start a go routine to kill task after max run time...
+	go func() {
+		time.Sleep(time.Second * time.Duration(task.Payload.MaxRunTime))
+		task.cancel("max run time (" + strconv.Itoa(task.Payload.MaxRunTime) + "s) exceeded")
+	}()
+	err = cmd.Wait()
+	// Reporting Task Result
+	// ---------------------
+	// If a task is malformed, the input is invalid, configuration is wrong, or
+	// the worker is told to shutdown by AWS before the the task is completed,
+	// it should be reported to the queue using `queue.reportException`.
+	if err != nil {
+		// If the task is unsuccessful, ie. exits non-zero, the worker should
+		// resolve it using `queue.reportFailed` (this implies test or build
+		// failure).
+		switch err.(type) {
+		case *exec.ExitError:
+			return task.reportFailed()
+		}
+
+		log.Printf("Command finished with error: %v", err)
+		return task.reportException()
+	}
+	// When the worker has completed the task successfully it should call
+	// `queue.reportCompleted`.
+	return task.reportCompleted()
+}
+
+func (task *TaskRun) prepEnvVars() {
+	for i, j := range task.Payload.Env {
+		fmt.Printf("Setting env var %v to '%v'\n", i, j)
+		os.Setenv(i, j)
```

Comment:
This can't be the  way of doing it... I'm sure one can set the environment variables of a subprocess to be different from current process.... Imagine a task overwrite worker config by mistake :)